components/style/properties/properties.mako.rs

```diff
@@ -1091,6 +1098,10 @@ impl PropertyDeclaration {
                     match input.try(|i| CSSWideKeyword::parse(context, i)) {
                         Ok(CSSWideKeyword::InheritKeyword) => {
                             % for sub_property in shorthand.sub_properties:
+                                if properties_seen.get_${property.ident}() {
```

Comment:
I'm wondering, why not doing this just above the `match input.try(..)` instead of here?

Also, do we set the correct bit when going through the `Err(())` path here (which is the non-css-wide-keyword one)?

It seems to me we don't. Please do that and add a test.